runtime_layer/vk_rt_layer_impl.cc

```diff
@@ -43,14 +56,71 @@ VkResult CreateImage(PFN_vkCreateImage            pfn,
                      VkImage*                     pImage)
 {
     // Remove VK_IMAGE_CREATE_FRAGMENT_DENSITY_MAP_OFFSET_BIT_QCOM flag
-    if ((pCreateInfo->flags & VK_IMAGE_CREATE_FRAGMENT_DENSITY_MAP_OFFSET_BIT_QCOM) != 0)
+    if (sRemoveImageFlagFDMOffset &&
+        ((pCreateInfo->flags & VK_IMAGE_CREATE_FRAGMENT_DENSITY_MAP_OFFSET_BIT_QCOM) != 0))
     {
-        // LOGI("Image %p CreateImage has the density map offset flag! \n", pImage);
+        LOGI("Image %p CreateImage has the density map offset flag! \n", pImage);
         const_cast<VkImageCreateInfo*>(pCreateInfo)
         ->flags &= ~VK_IMAGE_CREATE_FRAGMENT_DENSITY_MAP_OFFSET_BIT_QCOM;
     }
 
+    if (sRemoveImageFlagSubSampled &&
+        ((pCreateInfo->flags & VK_IMAGE_CREATE_SUBSAMPLED_BIT_EXT) != 0))
+    {
+        LOGI("Image %p CreateImage has the subsampled bit flag! \n", pImage);
+        const_cast<VkImageCreateInfo*>(pCreateInfo)->flags &= ~VK_IMAGE_CREATE_SUBSAMPLED_BIT_EXT;
+    }
+
     return pfn(device, pCreateInfo, pAllocator, pImage);
 }
 
+void CmdDrawIndexed(PFN_vkCmdDrawIndexed pfn,
+                    VkCommandBuffer      commandBuffer,
+                    uint32_t             indexCount,
+                    uint32_t             instanceCount,
+                    uint32_t             firstIndex,
+                    int32_t              vertexOffset,
+                    uint32_t             firstInstance)
+{
+    // Disable drawcalls with N index count
+    if (sEnableDrawcallFilter &&
+        ((indexCount == kNoDrawIndexCount) || (indexCount == kNoDrawIndexCount * 2)))
```

Comment:
this is to remove the drawcalls for visibility mask, on URP it is 84, on BiRP it is 2 drawcalls with 42

Reply:
this is just a prototype, ideally, there will be an UI for this, and the user can specify the count in the UI

Reply:
changed the const to kVisibilityMaskIndexCount to be more specific